perception/autoware_tensorrt_rtmdet/src/tensorrt_rtmdet.cpp

```diff
@@ -0,0 +1,396 @@
+    }
+    int index = b * image.cols * image.rows * 3;
+    // Copy into pinned memory
+    memcpy(
+      image_buf_h_.get() + index, &image.data[0],
+      image.cols * image.rows * 3 * sizeof(unsigned char));
+    b++;
+  }
+  // Copy into device memory
+  CHECK_CUDA_ERROR(cudaMemcpyAsync(
+    image_buf_d_.get(), image_buf_h_.get(),
+    images[0].cols * images[0].rows * 3 * batch_size * sizeof(unsigned char),
+    cudaMemcpyHostToDevice, *stream_));
+  // Preprocess on GPU
+  resize_bilinear_letterbox_nhwc_to_nchw32_batch_gpu(
+    input_d_.get(), image_buf_d_.get(), static_cast<int32_t>(input_width),
+    static_cast<int32_t>(input_height), 3, images[0].cols, images[0].rows, 3,
+    static_cast<int32_t>(batch_size), static_cast<float>(norm_factor_), *stream_);
+}
+
+void TrtRTMDet::preprocess(const std::vector<cv::Mat> & images)
+{
+  const auto batch_size = images.size();
+  auto input_dims = trt_common_->getBindingDimensions(0);
+  input_dims.d[0] = static_cast<int32_t>(batch_size);
+  trt_common_->setBindingDimensions(0, input_dims);
+
+  std::vector<cv::Mat> dst_images;
+  for (const auto & image : images) {
+    if (scale_width_ == -1 || scale_height_ == -1) {
+      scale_width_ = static_cast<float>(model_input_width_) / static_cast<float>(image.cols);
+      scale_height_ = static_cast<float>(model_input_height_) / static_cast<float>(image.rows);
+    }
+    cv::Mat dst_image;
+    cv::resize(
+      image, dst_image,
+      cv::Size(
+        static_cast<int32_t>(model_input_width_), static_cast<int32_t>(model_input_height_)));
+    dst_image.convertTo(dst_image, CV_32F);
+    dst_image -= cv::Scalar(103.53, 116.28, 123.675);
+    dst_image /= cv::Scalar(57.375, 57.12, 58.395);
+    dst_images.emplace_back(dst_image);
+  }
+
+  const auto chw_images = cv::dnn::blobFromImages(
+    dst_images, norm_factor_, cv::Size(), cv::Scalar(), false, false, CV_32F);
+
+  const auto data_length = chw_images.total();
+  input_h_.reserve(data_length);
+  const auto flat = chw_images.reshape(1, static_cast<int32_t>(data_length));
+  input_h_ = chw_images.isContinuous() ? flat : flat.clone();
+  CHECK_CUDA_ERROR(cudaMemcpy(
+    input_d_.get(), input_h_.data(), input_h_.size() * sizeof(float), cudaMemcpyHostToDevice));
+}
+
+bool TrtRTMDet::do_inference(
+  const std::vector<cv::Mat> & images, ObjectArrays & objects, cv::Mat & mask,
+  std::vector<uint8_t> & class_ids)
+{
+  if (!trt_common_->isInitialized()) {
+    return false;
+  }
+
+  if (use_gpu_preprocess_) {
+    preprocess_gpu(images);
+  } else {
+    preprocess(images);
+  }
+  return feedforward(images, objects, mask, class_ids);
+}
+
+bool TrtRTMDet::feedforward(
+  const std::vector<cv::Mat> & images, ObjectArrays & objects, cv::Mat & mask,
+  std::vector<uint8_t> & class_ids)
+{
+  std::vector<void *> buffers = {
+    input_d_.get(), out_dets_d_.get(), out_labels_d_.get(), out_masks_d_.get()};
+
+  trt_common_->enqueueV2(buffers.data(), *stream_, nullptr);
```

Comment:
@StepTurtle 
`enqueueV2` is removed from API since TensorRT 10.0.0 (source: https://docs.nvidia.com/deeplearning/tensorrt/archives/tensorrt-1000-ea/migration-guide/index.html ), so can you use `enqueueV3` instead of `enqueueV2` ?

Reply:
I used `enqueueV3` instead of `enqueueV2` but It depends [this commit](https://github.com/StepTurtle/autoware.universe/commit/7555b348b889fdd22983e8ef18aa7864dae4b65f). This commit adds some functional to tensorrt_common to handle `enqueueV3`.

What do you think about adding this to the `tensorrt_common` package? If it's appropriate to proceed this way, I can create a separate PR for this commit.

I saw preprocessor directives from: https://github.com/autowarefoundation/autoware.universe/issues/2330

related commit: https://github.com/autowarefoundation/autoware.universe/pull/8165/commits/602a5c8f0fd37273d8dd5708690869d589f712eb

Reply:
@amadeuszsz I need your advice. How do you think about it?

IMO, @StepTurtle's comment("What do you think about adding this to the tensorrt_common package?") is good idea.

Reply:
I would like to make an addition:

Currently, we expect to call `tensorrt_common::TrtCommon::setup()` after creating a `tensorrt_common::TrtCommon` instance. With my proposed changes, we would need to call the `tensorrt_common::TrtCommon::setupBindings()` function after buffer creation. If we agree on this approach, we should discuss and potentially update the setup functions to avoid requiring two separate calls for the setup process.

Reply:
@Shin-kyoto agree, `enqueueV3` should be supported in `tensorrt_common`.
@StepTurtle please, open separate PR for `enqueueV3` support in `tensorrt_common`. We currently considering to update TensorRT 10.0+ and its dependencies, therefore we will need to update all CUDA-related packages. Would be better if new packages could utilize `enqueueV3`.

Reply:
https://github.com/autowarefoundation/autoware.universe/issues/9017